week1_final/shopApp/internal/orders/store/store.go

```diff
@@ -0,0 +1,46 @@
+package store
+
+import (
+	"gorm.io/gorm"
+	"yash/shopApp/internal/orders/model"
+)
+
+// Store provides functionality for working with the database.
+type Store struct {
+	db *gorm.DB
+}
+
+// New will instantiate a new instance of Store.
+func New(db *gorm.DB) *Store {
+	db.AutoMigrate(&model.Order{})
```

Comment:
Why we are doing the AutoMigrate ??

Reply:
According to [this](https://gorm.io/docs/migration.html#Auto-Migration), it automatically generates the db schema based on the go structs. So, I figured it would be convenient to use it. However, now I also feel that we should not do this as it will modify the database according to the structs but it should not always be the case as database should be the source of truth. 

Let me know your thoughts on this.